sormas-backend/src/main/resources/sql/sormas_schema.sql

```diff
@@ -12958,4 +12958,24 @@ $$ LANGUAGE plpgsql;
 
 INSERT INTO schema_version (version_number, comment) VALUES (541, 'Update environment sample deletion dependencies #12887');
 
+-- 2024-02-09 Remove entity specific PERFORM_BULK_OPERATIONS user rights #10994
+DO $$
+  DECLARE
+     rec RECORD;
+  BEGIN
+       FOR rec IN SELECT id FROM userroles
+           LOOP
+               IF NOT EXISTS (SELECT 1 FROM userroles_userrights WHERE userrole_id = rec.id AND userright = 'PERFORM_BULK_OPERATIONS') THEN
+                  IF ((SELECT exists(SELECT userrole_id FROM userroles_userrights where userrole_id = rec.id and userright in ('PERFORM_BULK_OPERATIONS_CASE_SAMPLES','PERFORM_BULK_OPERATIONS_EVENT','PERFORM_BULK_OPERATIONS_EVENTPARTICIPANT','PERFORM_BULK_OPERATIONS_EXTERNAL_MESSAGES'))) = true) THEN
+                      INSERT INTO userroles_userrights(userrole_id, userright, sys_period) values (rec.id, 'PERFORM_BULK_OPERATIONS', tstzrange(now(), null));
+                  END IF;
+               END IF;
+               DELETE from userroles_userrights WHERE userrole_id = rec.id AND userright in ('PERFORM_BULK_OPERATIONS_CASE_SAMPLES', 'PERFORM_BULK_OPERATIONS_EVENT', 'PERFORM_BULK_OPERATIONS_EVENTPARTICIPANT','PERFORM_BULK_OPERATIONS_EXTERNAL_MESSAGES');
```

Comment:
I think this will remove ALL entries from the userroles_userrights table for the user; that's not good! You can just move the DELETE call out of the loop and write it like this:

```
DELETE FROM userroles_userrights WHERE userright IN (...);
```